### PR TITLE
Docs: Add balance examples.

### PR DESCRIPTION
**Pull request by Eideen**
Hi Dave
I have made a page with examples for balance of BTRFS, so it easier to understand how to use it.

### CODE REVIEW

File: Documentation/ch-balance-examples.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+Convert to miks of raid 1 and raid 10:
+***************************************
+If you have multi disk setup you can have diffrent profiles:
+
+* ``raid10`` for data
+* ``raid1c4`` for metadata and system. 
+
+.. code-block:: bash
+                
+        btrfs balance start -v -mconvert=raid1 -dconvert=raid10 $BTRFS_MOUNT
```
Comment (Zygo) [inline Documentation/ch-balance-examples.rst:70]: Could use `soft` here.

File: Documentation/ch-balance-examples.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+* https://lore.kernel.org/linux-btrfs/YnmItm%2FNW3eUcvsL@hungrycats.org
+
+Convert raid1 after mkfs with defaults:
+*****************************************
+If forgot to set the raid type creating the volume
+        
+.. code-block:: bash
+                
+        btrfs balance start -v convert=raid1 $BTRFS_MOUNT
```
[the remark this answers is not in the document]
Comment (Eideen) [reply Documentation/ch-balance-examples.rst:59]: I feel there is a trend to use soft for almost all balance tasks.

Being ´hard´ is currently default, when should not ´soft´ be used?

File: Documentation/ch-balance-examples.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+Convert to miks of raid 1 and raid 10:
+***************************************
+If you have multi disk setup you can have diffrent profiles:
```
[the remark this answers is not in the document]
Comment (Eideen) [reply Documentation/ch-balance-examples.rst:63]: I do agree with you. 

Will such information be more fitted in  `ch-balance-filters.rst` ?

File: Documentation/ch-balance-examples.rst [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+fix incomplet balance:
+***********************
+
+If the balance is interrupted ( reboot or stopped) during coverting to raid1.
+Only targets non `raid1` chunks.
+
+.. code-block:: bash
+
+        btrfs balance start -mconvert=raid1,soft $BTRFS_MOUNT
```
[the remark this answers is not in the document]
Comment (Eideen) [reply Documentation/ch-balance-examples.rst]: Correct, and system. 

Is the point about due to space constraints, valid for the example? 
Or more fit to be in the `ch-balance-filters.rst`.

> Before changing profiles, make sure there is enough unallocated space on
> existing drives to create new metadata block groups (for filesystems
> over 50GB, this is `1GB * (number_of_devices + 2))`.

File: Documentation/ch-balance-examples.rst [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+Convert to miks of raid 1 and raid 10:
+***************************************
+If you have multi disk setup you can have diffrent profiles:
+
+* ``raid10`` for data
+* ``raid1c4`` for metadata and system. 
+
+.. code-block:: bash
+                
+        btrfs balance start -v -mconvert=raid1 -dconvert=raid1 $BTRFS_MOUNT
```
[the remark this answers is not in the document]
Comment (Eideen) [reply Documentation/ch-balance-examples.rst]: It sound to me that `soft` should be the default for most user.

Do you have a example when not `soft` should be used?

**Comment by Eideen** [comment]
I have squashed the commits, and add changes to `ch-balance-filters.rst`, to reflect the input from @Zygo .

File: Documentation/ch-balance-examples.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (Zygo) [reply Documentation/ch-balance-examples.rst]: The space constraints always have to be observed.  A filesystem with unequal disk sizes and `raid1` profile can fill up with ordinary data writes if there isn't enough unallocated space.

The thing that makes balance different is that a new profile can be introduced, which changes the requirements for future allocations abruptly.

**Comment by kdave** [comment]
Thanks for the PR and comments, merged to devel. I've updated formatting and wording.

**Pull request closed without merging**